packages/db/tests/search/test-ReconciliationReportSearch.js

```diff
@@ -0,0 +1,259 @@
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test('ReconciliationReportSearch returns the correct response for a basic query', async (t) => {
+  const { knex } = t.context;
+  const dbSearch = new ReconciliationReportSearch({});
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results.length, 10);
+
+  const expectedResponse1 = {
+    name: 'fakeReconReport-1',
+    type: 'Granule Inventory',
+    status: 'Generated',
+    location: `s3://fakeBucket${t.context.reportBucket}/fakeKey${t.context.reportKey}`,
+    updatedAt: t.context.reconReportSearchTimestamp,
+    createdAt: t.context.reconReportSearchTimestamp,
+  }
+
+  const expectedResponse10 = {
+    name: 'fakeReconReport-10',
+    type: 'Granule Not Found',
+    status: 'Generated',
+    location: `s3://fakeBucket${t.context.reportBucket}/fakeKey${t.context.reportKey}`,
+    updatedAt: t.context.reconReportSearchTimestamp + 1,
+    createdAt: t.context.reconReportSearchTimestamp - 1,
+  }
+
+  t.deepEqual(response.results[0], expectedResponse1);
+  t.deepEqual(response.results[9], expectedResponse10);
+});
+
+test('ReconciliationReportSearch supports page and limit params', async (t) => {
+  const { knex } = t.context;
+  let queryStringParameters = {
+    limit: 25,
+    page: 2,
+  };
+  let dbSearch = new ReconciliationReportSearch({ queryStringParameters });
+  let response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 25);
+
+  queryStringParameters = {
+    limit: 10,
+    page: 5,
+  };
+  dbSearch = new ReconciliationReportSearch({ queryStringParameters });
+  response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 10);
+
+  queryStringParameters = {
+    limit: 10,
+    page: 11,
+  };
+  dbSearch = new ReconciliationReportSearch({ queryStringParameters });
+  response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 0);
+});
+
+test('ReconciliationReportSearch supports prefix search', async (t) => {
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 50,
+    prefix: 'fakeReconReport-1',
+  }
+  const dbSearch = new ReconciliationReportSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 11);
+  t.is(response.results?.length, 11);
+});
+
+test('ReconciliationReportSearch supports infix search', async (t) => {
+  const { knex } = t.context;
+  let queryStringParameters = {
+    limit: 50,
+    infix: 'conReport-2',
+  }
+  let dbSearch = new ReconciliationReportSearch({ queryStringParameters });
+  let response = await dbSearch.query(knex);
+  t.is(response.meta.count, 11);
+  t.is(response.results?.length, 11);
+  
+  // queryStringParameters = {
+  //   limit: 50,
+  //   infix: 'ending', // ending, status
```

Comment:
In [the api docs](https://nasa.github.io/cumulus-api/unreleased/#cumulus-api), it says for infix `includes search of the granuleId, status, pdrName, name, error, id and msg fields`. But this search I tried for infix in `status` didn't work. When I looked at what I copied from other search classes for `buildInfixPrefixQuery`, it only looks at "query.whereLike(`${this.tableName}.name`, `%${infix}%`)", so do I need to adjust the class, or is it only really supposed to look at `name`?

Reply:
as I understood this, infix query is for parsing something specific to the table in question (in this case name), but other, more generally present fields like granuleId are handled elsewhere in the buildSearch function (line 116 of BaseSearch.ts) by this.buildTermQuery mainly.  infix is just the one we need to tailor to this table

Reply:
so it's fine if we are supposed to customize/specify the field that the infix works for per table. But that isn't what our docs suggest, the docs seem to suggest you can do an includes search for any of the listed fields.
As an aside, one other issue I have with the docs for infix (maybe prefix also) and many of the other searches we are testing is that we don't have examples showing that some of those searches are possible.

Reply:
hm, I guess *is* that a behavior of the elasticsearch Search functionality we're imitating? in which case the docs were always wrong and should probably be fixed. or is that a functionality we've been failing to replicate so far in these moves over to PG

Reply:
I'm not sure, but it seems to me either way our docs are a bit off

Reply:
With ElasticSearch, we can easily match multiple fields. If we support the same infix/prefix search in postgres, the query can be complicated and performance is an concern, so we only support match one column per table.  If it becomes necessarily to match multiple columns, we can update in the future.